File: src/main/java/com/matag/admin/game/history/GameHistory.java
package com.matag.admin.game.history;

import com.matag.admin.game.game.GameType;
import com.matag.admin.game.game.GameUserResultType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class GameHistory {
  Long gameId;
  LocalDateTime startedTime;
  LocalDateTime finishedTime;
  GameType type;
  GameUserResultType result;
  String player1Name;
  String player1Options;
  String player2Name;
  String player2Options;
}

File: src/main/kotlin/com/matag/admin/game/history/GameHistory.kt
package com.matag.admin.game.history

import com.matag.admin.game.game.GameType
import com.matag.admin.game.game.GameUserResultType
import java.time.LocalDateTime

data class GameHistory(
    var gameId: Long? = null,
    var startedTime: LocalDateTime? = null,
    var finishedTime: LocalDateTime? = null,
    var type: GameType? = null,
    var result: GameUserResultType? = null,
    var player1Name: String? = null,
    var player1Options: String? = null,
    var player2Name: String? = null,
    var player2Options: String? = null
)
